Docs Authoring Pack

meganbradley opened the pull request:
Adding external facing documentation for the Docs Authoring Pack for VS Code.

opbld14 commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Contribute/TOC.md](https://github.com/meganbradley/Contribute/blob/master/Contribute/TOC.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/contribute/how-to-write-docs-auth-pack?branch=pr-en-us-7) |
[Contribute/how-to-write-docs-auth-pack.md](https://github.com/meganbradley/Contribute/blob/master/Contribute/how-to-write-docs-auth-pack.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/contribute/how-to-write-docs-auth-pack?branch=pr-en-us-7) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C4%5C9%5Cb7f7ecef-f7fb-e89c-41fa-c3f12167e1d5%5CPullRequest%5C201804092210131533-7%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=7yEesWK7VXTCcEUSijk9hz0S9cPtR6P9Lgipl2nURO0%3D&st=2018-04-09T22%3A06%3A00Z&se=2018-05-10T22%3A11%3A00Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

meganbradley commented:
@Jim-Parker @DuncanmaMSFT please review.

Jim-Parker commented:
@meganbradley is there a work item in VSTS for this?

meganbradley commented:
@Jim-Parker No - should I file one? Which VSTS?

Jim-Parker commented:
@meganbradley see the email I just sent, thanks

Jim-Parker commented:
One more question, is this for the internal or external guide?

meganbradley commented:
@Jim-Parker External - the extension is now  live in the VS Code Marketplace and can be installed by anyone.

Jim-Parker commented:
@meganbradley the article looks good to me. I'd suggest looping in Bill Wagner, since he's been very involved the organization of this TOC.

meganbradley commented:
@BillWagner can you review this PR and let me know if you want any changes? Thanks!

CC @Jim-Parker

BillWagner commented on Contribute/how-to-write-docs-auth-pack.md:
```diff
@@ -0,0 +1,89 @@
+|-------------|--------------------|----------------------|
+|Bold         |`formatBold`        |Formats text **bold**.|
+|Italic       |`formatItalic`      |Formats text *italic*.|
+|Code         |`formatCode`        |If one line or less is selected, formats text as `inline code`.<br><br>If multiple lines are selected, formats them as a fenced code block, and allows you to optionally select a programming language supported by OPS.<br><br>If no language is selected, inserts an empty fenced code block.|
```
3rd line, do you mean "If no lines are selected", instead of "if no language is selected"?

BillWagner commented on Contribute/how-to-write-docs-auth-pack.md:
```diff
@@ -0,0 +1,89 @@
+   
+    `"when": "editorTextFocus && editorLangId == 'markdown'"`
+
+    Your completed custom keybinding should look like this in keybindings.json:
```
Filenames should be italics:  *keybindings.json*

BillWagner commented on Contribute/how-to-write-docs-auth-pack.md:
```diff
@@ -0,0 +1,89 @@
+
+See [Keybindings](https://code.visualstudio.com/docs/getstarted/keybindings) in the VS Code docs for more information.
+
+## How to show the legacy "Gauntlet" toolbar
```
Gauntlet was never released publicly. Do we want to re-phrase some of this to avoid the use of a code name that many readers will not be familiar with? 

Or, if our long term plan is to deprecate the toolbar, should this section be moved to an internal document? /cc @Jim-Parker

DuncanmaMSFT replied:
We should remove reference to the Gauntlet term, and just explain that there is an optional toolbar that can be turned on. We should also be more specific that the setting in question is "markdown.showToolbar": true ... instead of the "the desired setting"

BillWagner commented on Contribute/TOC.md:
```diff
@@ -25,5 +25,6 @@
 ## Writing essentials 
 ### [Markdown](how-to-write-use-markdown.md)
 ### [Links](how-to-write-links.md)
+### [Docs Authoring Pack](how-to-write-docs-auth-pack.md)
```
Switch to sentence case: "Docs authoring pack"

meganbradley replied:
@BillWagner "Docs Authoring Pack" is a proper noun- it's the name of the extension pack (actually it was originally lower case but the GPM of VS Code asked us to capitalize it) - see https://marketplace.visualstudio.com/items?itemName=docsmsft.docs-authoring-pack

meganbradley commented:
Inline.

Thanks, I added to the PR.

From: Bill Wagner <notifications@github.com>
Sent: Monday, April 9, 2018 5:43 PM
To: MicrosoftDocs/Contribute <Contribute@noreply.github.com>
Cc: Megan Bradley <mbradley@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [MicrosoftDocs/Contribute] Docs Authoring Pack (#7)


@BillWagner requested changes on this pull request.

This looks great @meganbradley<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmeganbradley&data=02%7C01%7Cmbradley%40microsoft.com%7C5f948474db6149b785db08d59e7bffae%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636589177774472735&sdata=HaRLm9t8gA1KrZ%2B7p4xOzkCZIoGTVSW3L3NzD01PMbU%3D&reserved=0> (I've already installed the extension on my machines).

I had a few small comments and requested changes.

________________________________

In Contribute/how-to-write-docs-auth-pack.md<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoftDocs%2FContribute%2Fpull%2F7%23discussion_r180268705&data=02%7C01%7Cmbradley%40microsoft.com%7C5f948474db6149b785db08d59e7bffae%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636589177774482734&sdata=WG7iKTVhyDiKEgB769I0Yl2G%2Ft4aND9LXhABsIaieow%3D&reserved=0>:

> +

+## Prerequisites and assumptions

+

+To accurately insert relative links, images, and other embedded content with the Docs Markdown extension, you must have your VS Code workspace scoped to the root of your cloned OPS repo.

+

+Some syntax supported by the extension, such as alerts and snippets, are custom Markdown for OPS, and will not render correctly unless published via OPS.

+

+## How to use the Docs Markdown extension

+

+To access the Docs Markdown menu, type `ALT+M`. You can click or use up/down arrows to select the function you want, or type to start filtering, then hit `ENTER` when the function you want is highlighted in the menu. The following are available:

+

+|Function     |Command             |Description           |

+|-------------|--------------------|----------------------|

+|Bold         |`formatBold`        |Formats text **bold**.|

+|Italic       |`formatItalic`      |Formats text *italic*.|

+|Code         |`formatCode`        |If one line or less is selected, formats text as `inline code`.<br><br>If multiple lines are selected, formats them as a fenced code block, and allows you to optionally select a programming language supported by OPS.<br><br>If no language is selected, inserts an empty fenced code block.|

3rd line, do you mean "If no lines are selected", instead of "if no language is selected"? [Megan] No, it refers to what happens if you don’t select a programming language (see the previous sentence) – but this probably goes without saying so I’ll remove it.

________________________________

In Contribute/how-to-write-docs-auth-pack.md<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoftDocs%2FContribute%2Fpull%2F7%23discussion_r180269277&data=02%7C01%7Cmbradley%40microsoft.com%7C5f948474db6149b785db08d59e7bffae%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636589177774492742&sdata=f111zic5uPqMwes8BiHnQabO91cTi32Dk01u8jPouys%3D&reserved=0>:

> +|Include      |`insertInclude`   |Find a file to embed in the current file.|

+|Snippet      |`insertSnippet`   |Find a code snippet in the repo to embed in the current file.|

+|Video        |`insertVideo`     |Add an embedded video.|

+|Preview      |`previewTopic`    |Preview the active topic in a side-by-side window using the DocFX extension.  If the DocFX extension is not installed or is installed but disabled, the topic will not render.

+

+## How to assign keyboard shortcuts

+

+1. Type `CTRL+K` then `CTRL+S` to open the Keyboard Shortcuts list.

+1. Search for the command, such as `formatBold`, for which you want to create a custom keybinding.

+1. Click the plus that appears near the command name when you mouse over the line.

+1. After a new input box is visible, type the keyboard shortcut you want to bind to that particular command. For example, to use the common shortcut for bold, type `ctrl+b`.

+1. It's a good idea to insert a `when` clause into your keybinding, so it won't be available in files other than Markdown. To do this, open keybindings.json and insert the following line below the command name (be sure to add a comma between lines):

+

+    `"when": "editorTextFocus && editorLangId == 'markdown'"`

+

+    Your completed custom keybinding should look like this in keybindings.json:

Filenames should be italics: keybindings.json [Megan] Fixed.

________________________________

In Contribute/how-to-write-docs-auth-pack.md<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoftDocs%2FContribute%2Fpull%2F7%23discussion_r180269613&data=02%7C01%7Cmbradley%40microsoft.com%7C5f948474db6149b785db08d59e7bffae%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636589177774492742&sdata=E533m2DfXLFpvQFd6aoMLIXAM9XwrOJ2boKNxe%2Brymk%3D&reserved=0>:

> +    ```json

+    // Place your key bindings in this file to overwrite the defaults

+    [

+        {

+            "key": "ctrl+b",

+            "command": "formatBold",

+            "when": "editorTextFocus && editorLangId == 'markdown'"

+        }

+    ]

+    ```

+

+1. Save keybindings.json.

+

+See [Keybindings](https://code.visualstudio.com/docs/getstarted/keybindings) in the VS Code docs for more information.

+

+## How to show the legacy "Gauntlet" toolbar

Gauntlet was never released publicly. Do we want to re-phrase some of this to avoid the use of a code name that many readers will not be familiar with?

Or, if our long term plan is to deprecate the toolbar, should this section be moved to an internal document? /cc @Jim-Parker<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FJim-Parker&data=02%7C01%7Cmbradley%40microsoft.com%7C5f948474db6149b785db08d59e7bffae%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636589177774502759&sdata=h4geSrcy3nL84XvDF5%2Bj3YPfYK%2BOCZ56SrxnuI4RZpQ%3D&reserved=0> [Megan] It can go – I removed it.

________________________________

In Contribute/TOC.md<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoftDocs%2FContribute%2Fpull%2F7%23discussion_r180269779&data=02%7C01%7Cmbradley%40microsoft.com%7C5f948474db6149b785db08d59e7bffae%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636589177774502759&sdata=a2TaYJh3p7tA%2FgTzDPngD4YCIxGd6ZpfOu%2FiR9mVFK4%3D&reserved=0>:

> @@ -25,5 +25,6 @@

 ## Writing essentials

 ### [Markdown](how-to-write-use-markdown.md)

 ### [Links](how-to-write-links.md)

+### [Docs Authoring Pack](how-to-write-docs-auth-pack.md)

Switch to sentence case: "Docs authoring pack" [Megan] It’s a proper noun – the name of the extension pack is capitalized.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoftDocs%2FContribute%2Fpull%2F7%23pullrequestreview-110660681&data=02%7C01%7Cmbradley%40microsoft.com%7C5f948474db6149b785db08d59e7bffae%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636589177774512762&sdata=UVXVJn0GDzdIEBu0xBAmK7GTq5BLPdbgZLsk3x4xRdQ%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAQgKfTbM5dF5t_Fffvc2mRR1RG8u9Lstks5tnAAOgaJpZM4TNTh0&data=02%7C01%7Cmbradley%40microsoft.com%7C5f948474db6149b785db08d59e7bffae%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636589177774522766&sdata=lT%2FcAKYK2czfNx%2B24qZJah97XFRNv6rmkjpVDOCYcmU%3D&reserved=0>.

opbldsb3 commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Contribute/TOC.md](https://github.com/meganbradley/Contribute/blob/master/Contribute/TOC.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/contribute/how-to-write-docs-auth-pack?branch=pr-en-us-7) |
[Contribute/how-to-write-docs-auth-pack.md](https://github.com/meganbradley/Contribute/blob/master/Contribute/how-to-write-docs-auth-pack.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/contribute/how-to-write-docs-auth-pack?branch=pr-en-us-7) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C4%5C10%5Cb7f7ecef-f7fb-e89c-41fa-c3f12167e1d5%5CPullRequest%5C201804101646282657-7%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=w%2F7%2FmrsOLvousyRNi9kb1wnS%2F54mkYV32cdagSlEwwU%3D&st=2018-04-10T16%3A43%3A01Z&se=2018-05-11T16%3A48%3A01Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

MonicaRush commented:
@meganbradley @BillWagner @Jim-Parker Bill has signed off. Megan and Jim, can you add "#sign-off" too when you are ready? Thanks -- Monica

meganbradley commented:
#sign-off

Jim-Parker commented:
@meganbradley @MonicaRush for the external guide #sign-off
